packages/org.openapis.v3/Operation.pkl

```diff
@@ -0,0 +1,112 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// Describes a single API operation on a path.
+module org.openapis.v3.Operation
+
+import "OpenAPI.pkl"
+import "ExternalDocs.pkl"
+import "Parameter.pkl"
+import "Reference.pkl"
+import "RequestBody.pkl"
+import "Response.pkl"
+import "Security.pkl"
+import "Server.pkl"
+import "PathItem.pkl"
+import "HTTPResponse.pkl"
+
+/// A list of tags for API documentation control.
+///
+/// Tags can be used for logical grouping of operations by resources or any other qualifier.
+tags: Listing<String>(isDistinct)?
+
+/// A short summary of what the operation does.
+summary: String?
+
+/// A verbose explanation of the operation behavior. CommonMark syntax MAY be used for rich text representation.
+@SourceCode { language = "Markdown" }
+description: String?
+
+/// Additional external documentation for this operation.
+externalDocs: ExternalDocs?
+
+/// Unique string used to identify the operation.
+///
+/// The id MUST be unique among all operations described in the API. The operationId value is case-sensitive. Tools
+/// and libraries MAY use the operationId to uniquely identify an operation, therefore, it is RECOMMENDED to follow
+/// common programming naming conventions.
+operationId: String?
+
+/// A list of parameters that are applicable for this operation.
+///
+/// If a parameter is already defined at the Path Item, the new definition will override it but can never remove it.
+/// The list MUST NOT include duplicated parameters. A unique parameter is defined by a combination of a name and
+/// location. The list can use the Reference Object to link to parameters that are defined at the OpenAPI Object's
+/// components/parameters.
+parameters: Mapping<String, Parameters>?
+
+// follow the uniqueness standard for parameters set forth by the spec
+typealias Parameters = Listing<*Parameter|Reference>(
+    isDistinctBy((it) ->
+      if (it is Parameter) "\(it.name)\(it.`in`)" else it.`$ref`)
+    || throw("params must be unique by name and location"))
+
+/// The request body applicable for this operation.
+///
+/// The requestBody is only supported in HTTP methods where the HTTP 1.1 specification RFC7231 has explicitly defined
+/// semantics for request bodies. In other cases where the HTTP spec is vague, requestBody SHALL be ignored by
+/// consumers.
+requestBody: *RequestBody|Reference?
+
+/// The list of possible responses as they are returned from executing this operation.
+///
+/// Any HTTP status code can be used as the property name, but only one property per code, to describe the expected
+/// response for that HTTP statuscode. You should use const values of _HTTPResponse.Code_ for this, eg.
```

Comment:
I actually don't think it's that necessary to use them. Even if you define the strings literally, the IDE will provide nice code completion for you.

BTW: this can be a member link; you get code navigation that way.

```suggestion
/// response for that HTTP statuscode. You should can use values of [HTTPResponse] for this, eg.
```

Reply:
Bit of a slip of the tongue there. I agree.